### PR TITLE
✨ Add .NET Aspire (#362)

### PR DESCRIPTION
**Pull request by danielmackay**
﻿> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ https://github.com/SSWConsulting/SSW.CleanArchitecture/issues/362

> 2. What was changed?

✏️ Replaced docker compose with Aspire

> 3. Did you do pair or mob programming?

✏️ No

### CODE REVIEW

File: SSW.CleanArchitecture.sln [hunk 1]
```diff
@@ -39,6 +37,8 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tools", "tools", "{7AFFF91C
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Database", "tools\Database\Database.csproj", "{08A4FCE5-D49A-4FC1-BB0B-3206326157BD}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "AppHost", "tools\AppHost\AppHost.csproj", "{874CCCC1-3F73-4DA9-BA2B-89EE6C7C4B8E}"
```
Comment (MattParkerDev) [inline SSW.CleanArchitecture.sln:40]: I'd consider naming this Aspire.

File: tests/WebApi.IntegrationTests/Common/Fixtures/DatabaseContainer.cs [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
-            ConnectionString = _container.GetConnectionString();
-        }
+        await StartWithRetry();
+        ConnectionString = _container.GetConnectionString();
     }
 
-    public Task DisposeAsync() => _container?.StopAsync() ?? Task.CompletedTask;
+    private async Task StartWithRetry()
+    {
+        // NOTE: For some reason the container sometimes fails to start up.  Add in a retry to protect against this
```
Comment (MattParkerDev) [inline tests/WebApi.IntegrationTests/Common/Fixtures/DatabaseContainer.cs:31]: 😂

File: tools/AppHost/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
+      "applicationUrl": "http://localhost:15231",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development",
+        "DOTNET_ENVIRONMENT": "Development",
+        "DOTNET_DASHBOARD_OTLP_ENDPOINT_URL": "http://localhost:19133",
+        "DOTNET_RESOURCE_SERVICE_ENDPOINT_URL": "http://localhost:20264"
+      }
```
Comment (MattParkerDev) [inline tools/AppHost/Properties/launchSettings.json:26]: This could be deleted.

**Review by MattParkerDev — APPROVED** [review]

File: SSW.CleanArchitecture.sln (hunk 1 above)
Comment (danielmackay) [reply SSW.CleanArchitecture.sln:40]: As per our conversation, your comment makes sense, but most of the examples I've seen have this left as the 'AppHost'.  I think keeping it named like aligns with other projects I've seen.

File: tools/AppHost/AppHost.csproj [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <Sdk Name="Aspire.AppHost.Sdk" Version="9.0.0-rc.1.24511.1"/>
```
[the remark this answers is not in the document]
Comment (danielmackay) [reply tools/AppHost/AppHost.csproj:3]: Aspire uses a slightly different release cadence to .NET.  RC 1 is the latest.

File: tools/AppHost/Properties/launchSettings.json (hunk 3 above)
Comment (danielmackay) [reply tools/AppHost/Properties/launchSettings.json:26]: Will leave it for now, but perhaps remove in future

**Pull request merged**